ob-deno.el

```diff
@@ -97,22 +97,54 @@ which can be specified by VALUES."
     (format "--allow-%s" allow-param)))
 
 (defun ob-deno-read (results)
-  "Convert RESULTS into an appropriate elisp value.
-If RESULTS look like a table, then convert them into an
-Emacs-lisp table, otherwise return the results as a string."
-  (org-babel-read
-   (if (and (stringp results)
-	    (string-prefix-p "[" results)
-	    (string-suffix-p "]" results))
-       (org-babel-read
-        (concat "'"
-                (replace-regexp-in-string
-                 "\\[" "(" (replace-regexp-in-string
-                            "\\]" ")" (replace-regexp-in-string
-                                       ",[[:space:]]" " "
-				       (replace-regexp-in-string
-					"'" "\"" results))))))
-     results)))
+  "Convert RESULTS into an Org-mode table recognized by Org Babel.
+Handles JSON arrays of objects, arrays, primitives, and single objects.
+Preserves key order and empty rows."
+  (let ((json-array (json-parse-string results)))
+    (cond
+     ;; Case 1: Single Object (hash table)
+     ((hash-table-p json-array)
+      (let ((keys '())
+            (row '()))
+        ;; Collect keys and build row in insertion order
+        (maphash (lambda (key value)
+                   (setq keys (append keys (list key)))
+                   (setq row (append row (list (if value (format "%s" value) "")))))
+                 json-array)
+        (append (list keys 'hline) (list row))))
+
+     ;; Case 2: Array of Objects (vector of hash tables)
+     ((and (vectorp json-array) (hash-table-p (aref json-array 0)))
```

Comment:
Would it be better to check every element so that we can have fever number of mismatches?

```suggestion
     ((and (vectorp json-array) (seq-every-p #'hash-table-p json-array))
```

Reply:
🤔 what do you think the user is expecting if they have a heterogenous array of objects?

Reply:
My reasoning is that if you have an heterogeneous array then there isn't a common way to visualize it so you would *probably* want to see the raw output rather than a half formatted output. If the output being heterogeneous is a mistake, then they can simply spot it by looking at the output rather than having a random error.